students/aaron/lesson01/assignment/test_unit.py

```diff
@@ -0,0 +1,186 @@
+        Tests main.main_menu()
+        """
+        self.assertEqual(main.main_menu('1'), main.add_new_item)
+        self.assertEqual(main.main_menu('2'), main.item_info)
+        self.assertEqual(main.main_menu('q'), main.exit_program)
+        with patch('builtins.input', lambda x: '1'):
+            menu = main.main_menu()
+            self.assertEqual(menu, main.add_new_item)
+    
+    def test_add_new_item(self):
+        """
+        Tests main.add_new_item()
+        """
+        input_vars = ['3', 'Car', '1', 'n', 'n']
+        inventory = {}
+        with patch('builtins.input', side_effect=input_vars):
+            main.add_new_item(inventory)
+        self.assertEqual(inventory['3'],
+                         {
+                             'product_code': '3',
+                             'description': 'Car',
+                             'market_price': 24,
+                             'rental_price': '1'
+                         })
+
+    def test_add_new_furniture(self):
+        """
+        Tests main.add_new_item() with a furniture item
+        """
+        input_vars = ['4', 'Rug', '1', 'y', 'Berber', 's']
+        inventory = {}
+        with patch('builtins.input', side_effect=input_vars):
+            main.add_new_item(inventory)
+        self.assertEqual(inventory['4'],
+                         {
+                             'product_code': '4',
+                             'description': 'Rug',
+                             'market_price': 24,
+                             'rental_price': '1',
+                             'material': 'Berber',
+                             'size': 's'
+                         })
+
+    def test_add_new_electric(self):
+        """
+        Tests main.add_new_item() with an electric appliance
+        """
+        input_vars = ['5', 'Shaver', '1', 'n', 'y', 'Norelco', '110']
+        inventory = {}
+        with patch('builtins.input', side_effect=input_vars):
+            main.add_new_item(inventory)
+        self.assertEqual(inventory['5'],
+                         {
+                             'product_code': '5',
+                             'description': 'Shaver',
+                             'market_price': 24,
+                             'rental_price': '1',
+                             'brand': 'Norelco',
+                             'voltage': '110'
+                         })
+
+    def test_item_info(self):
+        """
+        Tests main.item_info()
+        """
+        input_vars = ['6', '99']
+        test_item = {'product_code': '6',
+                     'description': 'item',
+                     'market_price': 24,
+                     'rental_price': '1'}
+        inventory = {'6': test_item}
+        with patch('builtins.input', side_effect=input_vars):
+            item = main.item_info(inventory)
+            bogus_item = main.item_info(inventory)
+        self.assertEqual(item, test_item)
+        self.assertEqual(bogus_item, None)
```

Comment:
Nice. You are testing that bogus item is NOT there.

Reply:
Thanks!